extensions/ext/GLSL_EXT_integer_dot_product.txt

```diff
@@ -0,0 +1,288 @@
+Name
+
+    EXT_integer_dot_product
+
+Name Strings
+
+    GL_EXT_integer_dot_product
+
+Contact
+
+    Jeff Bolz, NVIDIA (jbolz 'at' nvidia.com)
+
+Contributors
+
+    Jeff Bolz, NVIDIA (jbolz 'at' nvidia.com)
+    Kevin Petit, Arm Ltd. (kevin.petit 'at' arm.com)
+    Tobias Hector, AMD
+    Graeme Leese, Broadcom Corporation
+    Georg Lehmann
+
+Status
+
+    Complete.
+
+Version
+
+    Last Modified:      October 3, 2022
+    Revision:           1
+
+Dependencies
+
+    This extension can be applied to OpenGL GLSL versions 4.50
+    (#version 450) and higher.
+
+    This extension can be applied to OpenGL ES ESSL versions 3.20
+    (#version 320) and higher.
+
+    This extension is written against the OpenGL Shading Language
+    Specification, version 4.60, dated July 23, 2017.
+
+    This extension interacts with
+    GL_EXT_shader_explicit_arithmetic_types_int8,
+    GL_EXT_shader_explicit_arithmetic_types_int16 and
+    GL_EXT_shader_explicit_arithmetic_types_int64.
+
+Overview
+
+    This extension adds mixed-precision integer dot product instructions.
+
+    Mapping to SPIR-V
+    -----------------
+
+    For informational purposes (non-normative), the following is an
+    expected way for an implementation to map GLSL constructs to SPIR-V
+    constructs, the SPIR-V capabilities they require, and the Vulkan
+    feature bits those capabilities require.
+
+    All functions introduced in this extension require support for
+    the shaderIntegerDotProduct Vulkan feature and DotProductKHR SPIR-V
+    capability.
+
+    The dotPacked4x8EXT and dotPacked4x8AccSatEXT functions require support for
+    the DotProductInput4x8BitPackedKHR SPIR-V capability.
+
+    All functions with ivec8 or uvec8 input vectors require support for the
+    DotProductInput4x8BitKHR SPIR-V capability.
```

Comment:
The SPIR-V for this is specific to 4-component vectors, so there seems to be a mismatch. Should one or other be updated to match?

Reply:
The GLSL spec should say `i8vec4` and `u8vec4`.